cmd/cortex/main_test.go

```diff
@@ -76,7 +76,22 @@ func TestFlagParsing(t *testing.T) {
 
 		"root level configuration option specified as an empty node in YAML": {
 			yaml:          "querier:",
-			stderrMessage: "the Querier configuration in YAML has been specified as an empty YAML node",
+			stderrMessage: "the querier configuration in YAML has been specified as an empty YAML node",
+		},
+
+		"root level configuration option specified as an empty object node in YAML": {
+			yaml:          "querier: {}",
+			stderrMessage: "the querier configuration in YAML has been specified as an empty YAML node",
+		},
+
+		"root level configuration option specified as a null node in YAML": {
+			yaml:          "querier: null",
+			stderrMessage: "the querier configuration in YAML has been specified as an empty YAML node",
+		},
+
+		"root level configuration option specified with all zero values": {
+			yaml:           "flusher: { exit_after_flush: false }",
+			stderrExcluded: "empty YAML node",
```

Comment:
Hum.. why this should be invalid?

Reply:
This is a test case that fails under the old validation logic, but should not.

Reply:
There are some more details in the linked issue.

Reply:
oh ok..
that make sense i guess...